Bottom: a76bd5aa6fdbc39f128bf67e79a571f5cff6ce84
Top:    d04f6576c141d1355c09910b6249287e53eb33ef
Author: Peter Clifton <pcjc2@cam.ac.uk>
Date:   2010-12-20 14:02:28 +0000

Temp fixes and diagnostics


---

diff --git a/src/polygon1.c b/src/polygon1.c
index 42ffdeb..7dae18f 100644
--- a/src/polygon1.c
+++ b/src/polygon1.c
@@ -1123,7 +1123,11 @@ label_contour (PLINE * a)
       assert (label == INSIDE || label == OUTSIDE);
       LABEL_NODE (cur, label);
     }
-  while ((cur = cur->next) != first_labelled);
+  while ((cur = cur->next) != first_labelled &&
+	 !(cur->next == &a->head && first_labelled == NULL));
+
+  if (first_labelled == NULL)
+    fprintf (stderr, "OOPS, DIDN'T LABEL ANYTHING!\n");
 #ifdef DEBUG_ALL_LABELS
   print_labels (a);
   DEBUGP ("\n\n");
diff --git a/src/sweep.h b/src/sweep.h
new file mode 100644
index 0000000..e69de29
